plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -277,6 +278,16 @@ protected boolean isTableLockNeeded(ConnectorSession session)
         return isBulkCopyForWrite(session) && isBulkCopyForWriteLockDestinationTable(session);
     }
 
+    @Override
+    protected void verifyTableName(DatabaseMetaData databaseMetadata, String tableName)
+            throws SQLException
+    {
+        // SQL Server truncates table name to 128 chars silently when renaming a table
```

Comment:
nit: The comment is slightly confusing because it talks about 128 but does comparison with `databaseMetadata.getMaxTableNameLength()`.